src/common/Kokkos_ArithTraits.hpp

```diff
@@ -225,6 +225,230 @@ KOKKOS_FORCEINLINE_FUNCTION IntType intPowUnsigned(const IntType x,
+  }                                                                            \
+  static FUNC_QUAL val_type pow(const val_type x, const val_type y) {          \
+    return Kokkos::pow(x, y);                                                  \
+  }                                                                            \
+  static FUNC_QUAL val_type pow(const val_type x, const mag_type y) {          \
+    return Kokkos::pow(x, y);                                                  \
+  }                                                                            \
+  static FUNC_QUAL val_type pow(const mag_type x, const val_type y) {          \
+    return Kokkos::pow(x, y);                                                  \
+  }                                                                            \
+  static FUNC_QUAL val_type sqrt(const val_type x) {                           \
+    return ::Kokkos::sqrt(x);                                                  \
+  }                                                                            \
+  static FUNC_QUAL val_type exp(const val_type x) { return Kokkos::exp(x); }   \
+  static FUNC_QUAL val_type log(const val_type x) { return Kokkos::log(x); }   \
+  static FUNC_QUAL val_type log10(const val_type x) {                          \
+    return Kokkos::log10(x);                                                   \
+  }                                                                            \
+  static FUNC_QUAL val_type sin(const val_type x) { return Kokkos::sin(x); }   \
+  static FUNC_QUAL val_type cos(const val_type x) { return Kokkos::cos(x); }   \
+  static FUNC_QUAL val_type tan(const val_type x) { return Kokkos::tan(x); }   \
+  static FUNC_QUAL val_type sinh(const val_type x) { return Kokkos::sinh(x); } \
+  static FUNC_QUAL val_type cosh(const val_type x) { return Kokkos::cosh(x); } \
+  static FUNC_QUAL val_type tanh(const val_type x) { return Kokkos::tanh(x); } \
+  static FUNC_QUAL val_type asin(const val_type x) { return Kokkos::asin(x); } \
+  static FUNC_QUAL val_type acos(const val_type x) { return Kokkos::acos(x); } \
+  static FUNC_QUAL val_type atan(const val_type x) { return Kokkos::atan(x); } \
+  static FUNC_QUAL bool isnaninf(const val_type& x) {                          \
+    return isNan(x) || isInf(x);                                               \
+  }                                                                            \
+  static FUNC_QUAL mag_type magnitude(const val_type x) { return abs(x); }     \
+  static FUNC_QUAL val_type conjugate(const val_type x) { return conj(x); }    \
+  static FUNC_QUAL val_type squareroot(const val_type x) { return sqrt(x); }   \
+  static FUNC_QUAL mag_type eps() { return epsilon(); }
+
+#define KOKKOSKERNELS_SIGNED_ABS                          \
+  static KOKKOS_FUNCTION mag_type abs(const val_type x) { \
+    return Kokkos::abs(x);                                \
```

Comment:
Just making sure you understand this will involve promotion to `int` for the smaller integer types.
If you are using it for signed integers you might as well used it for the unsigned ones.